platform/android/export/export.cpp

```diff
@@ -2113,43 +2162,37 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			//write
 
 			if (file == "AndroidManifest.xml") {
-
 				_fix_manifest(p_preset, data, p_flags & (DEBUG_FLAG_DUMB_CLIENT | DEBUG_FLAG_REMOTE_DEBUG));
 			}
 
 			if (file == "resources.arsc") {
-
 				_fix_resources(p_preset, data);
 			}
 
-			if (file == "res/drawable-nodpi-v4/icon.png") {
-				bool found = false;
-				for (uint64_t i = 0; i < sizeof(launcher_icons) / sizeof(launcher_icons[0]); ++i) {
-					String icon_path = String(p_preset->get(launcher_icons[i].option_id)).strip_edges();
-					if (icon_path != "" && icon_path.ends_with(".png")) {
-						FileAccess *f = FileAccess::open(icon_path, FileAccess::READ);
-						if (f) {
-							data.resize(f->get_len());
-							f->get_buffer(data.ptrw(), data.size());
-							memdelete(f);
-							found = true;
-							break;
-						}
-					}
-				}
-				if (!found) {
-
-					String appicon = ProjectSettings::get_singleton()->get("application/config/icon");
-					if (appicon != "" && appicon.ends_with(".png")) {
-						FileAccess *f = FileAccess::open(appicon, FileAccess::READ);
-						if (f) {
-							data.resize(f->get_len());
-							f->get_buffer(data.ptrw(), data.size());
-							memdelete(f);
-						}
-					}
-				}
+			String project_icon_path = ProjectSettings::get_singleton()->get("application/config/icon");
```

Comment:
Can you move this outside of the `while` loop since it doesn't change.

Reply:
For sure.